Mac-App/Mac-App/StateObserver/StateObserver.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  StateObserver.swift
+//  Mac-App
+//
+//  Created by Daniel GARCÍA on 03.05.20.
+//  Copyright © 2020 Acphut Werkstatt. All rights reserved.
+//
+
+import Combine
+import ReSwift
+
+final class StateObserver<State: StateType & Equatable>: StoreSubscriber {
+    typealias StoreSubscriberStateType = State
+
+    @Published
+    private(set) var currentState: State!
```

Comment:
Unwrapping here as this class is not mean to be used in any other way. I wish I could mark `newState` as private but doing so deviates from the `StoreSubscriber` protocol conformance.